tests/functional/adapter/test_timestamps.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+from dbt.tests.adapter.utils import test_timestamps
+
+_MODEL_CURRENT_TIMESTAMP = """
+SELECT {{current_timestamp()}} as current_timestamp,
+       {{current_timestamp_in_utc_backcompat()}} as current_timestamp_in_utc_backcompat,
+       {{current_timestamp_backcompat()}} as current_timestamp_backcompat
+"""
+
+class TestCurrentTimestampSnowflake(test_timestamps.TestCurrentTimestamps):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"get_current_timestamp.sql": _MODEL_CURRENT_TIMESTAMP}
+
+    @pytest.fixture(scope="class")
+    def expected_schema(self):
+        return {
+                "CURRENT_TIMESTAMP": "TIMESTAMP_TZ",
+                "CURRENT_TIMESTAMP_IN_UTC_BACKCOMPAT": "TIMESTAMP_NTZ",
+                "CURRENT_TIMESTAMP_BACKCOMPAT": "TIMESTAMP_NTZ",
+            }
```

Comment:
It doesn't make a functional difference, but do you want to lowercase this for consistency?